pkg/codegen/pcl/rewrite_convert.go

```diff
@@ -102,9 +102,14 @@ func rewriteConversions(x model.Expression, to model.Type) (model.Expression, bo
 			var traverser hcl.Traverser
 			if lit, ok := item.Key.(*model.LiteralValueExpression); ok {
 				traverser = hcl.TraverseIndex{Key: lit.Value}
+			} else if temp, isTemp := item.Key.(*model.TemplateExpression); isTemp && len(temp.Parts) == 1 {
+				if literal, isLiteral := temp.Parts[0].(*model.LiteralValueExpression); isLiteral {
+					traverser = hcl.TraverseIndex{Key: literal.Value}
+				}
```

Comment:
What happens in the else case? Do we intend to leave `traverser` uninitialized?

Reply:
I assumed that doesn't happen but just in case I guess we can initialize it to `model.MakeTraverser(model.StringType)` but I am not sure how that works when it doesn't know which value to use as a traverser

Reply:
If you don't think that a `model.MakeTraverser(model.StringType)` makes sense, should we explicitly `panic`? If it does make sense, should we move the default case to when we initialize `traverser`? Then we can remove both `else` final cases. I think this communicates that this is the default case more clearly?

Reply:
@iwahbe I've updated the PR with a default initialization here, but I'm not sure if it's correct still. I'd rather it be incorrect and proceed however, in case it can later recover with a diagnostic, than panic.

Tests updated and rebased on #10043 without duplicating the .pp files.